### ISSUE TITLE
[bug] mapped pages during the `map_page` and `content_harvest` tasks periodically stop running, and need to be restarted--seemingly a timeout issue

### ISSUE THREAD

**Issue by christinklez** [ucldc/rikolti#839]
By "the task is prone to stop running several times along the way," -- this tends to happen with larger collections that have many task pages. Usually during the `map_page` or `content_harvest` task.

Sometime while it's running a batch of `map_page` or `content_harvest` task pages, it will stop running and turn red. It will have failed on a particular page (but it gets tricky to find the particular page). The entire job will then stop. Once stopped, I can retry the "failed tasks" only, and it will retry only the red task pages.

**Here's an example:**
- Run ID: manual__2024-03-27T00:02:50+00:00
   - During content_harvest mapped page 8
   - [Link to airflow grid](https://7a8067cb-3b99-477e-a883-7e311175a9b4.c3.us-west-2.airflow.amazonaws.com/dags/harvest_collection/grid?num_runs=365&dag_run_id=manual__2024-03-27T00%3A02%3A50%2B00%3A00&task_id=content_harvesting.content_harvest&tab=mapped_tasks&map_index=8)
   - [Permalink to log](https://7a8067cb-3b99-477e-a883-7e311175a9b4.c3.us-west-2.airflow.amazonaws.com/log?dag_id=harvest_collection&task_id=content_harvesting.content_harvest&execution_date=2024-03-27T00%3A02%3A50%2B00%3A00&map_index=8)

In this example, you can see there were two attempts. The first attempt error'd out and failed (along with other pages that failed). Later, when the entire `content_harvest` job failed, I clicked on the content_harvest job (and not a specific task page) and clicked "Clear" > "Only Failed" to rerun the failed pages. The second attempt was successful.

From log 1:
```
[2024-03-27, 00:31:38 UTC] {{taskinstance.py:1308}} INFO - Starting attempt 1 of 1
[2024-03-27, 00:31:38 UTC] {{taskinstance.py:1327}} INFO - Executing <Mapped(ContentHarvestEcsOperator): content_harvesting.content_harvest> on 2024-03-27 00:02:50+00:00
[2024-03-27, 00:31:38 UTC] {{standard_task_runner.py:57}} INFO - Started process 28632 to run task
[2024-03-27, 00:31:38 UTC] {{standard_task_runner.py:84}} INFO - Running: ['airflow', 'tasks', 'run', 'harvest_collection', 'content_harvesting.content_harvest', 'manual__2024-03-27T00:02:50+00:00', '--job-id', '31319', '--raw', '--subdir', 'DAGS_FOLDER/rikolti/dags/harvest_dag.py', '--cfg-path', '/tmp/tmp4oc5e78d', '--map-index', '8']
[2024-03-27, 00:31:38 UTC] {{standard_task_runner.py:85}} INFO - Job 31319: Subtask content_harvesting.content_harvest
[2024-03-27, 00:31:48 UTC] {{local_task_job_runner.py:291}} WARNING - State of this instance has been externally set to scheduled. Terminating instance.
[2024-03-27, 00:31:48 UTC] {{process_utils.py:131}} INFO - Sending Signals.SIGTERM to group 28632. PIDs of all processes in the group: [28632]
[2024-03-27, 00:31:48 UTC] {{process_utils.py:86}} INFO - Sending the signal Signals.SIGTERM to group 28632
[2024-03-27, 00:31:48 UTC] {{process_utils.py:79}} INFO - Process psutil.Process(pid=28632, status='terminated', exitcode=<Negsignal.SIGTERM: -15>, started='00:31:38') (28632) terminated with exit code Negsignal.SIGTERM
```

***

Initial issue described below is actually a non-issue.

## Expected
When a mapped task does not run, the status should be marked as failed, and represented by a red square.

## What's happening
When a mapped task does not run, the status displays as a success, and is represented by a green square. This occurred during a run on March 23rd.

## Example from March 23
- [ ] 27143, mapped task 749 - [link to content_harvest grid 749](https://7a8067cb-3b99-477e-a883-7e311175a9b4.c3.us-west-2.airflow.amazonaws.com/dags/harvest_collection/grid?num_runs=100&dag_run_id=manual__2024-03-05T21%3A18%3A43%2B00%3A00&task_id=content_harvesting.content_harvest&tab=mapped_tasks&map_index=749)
   - Switch to the "details" tab, and you will see the green success indicator
   - Run ID: `manual__2024-03-05T21:18:43+00:00`
   - Mapper: omeka.nothumb
   - content_harvest mapped page permalink: [749](https://7a8067cb-3b99-477e-a883-7e311175a9b4.c3.us-west-2.airflow.amazonaws.com/log?dag_id=harvest_collection&task_id=content_harvesting.content_harvest&execution_date=2024-03-05T21%3A18%3A43%2B00%3A00&map_index=749)

What concerns me is that I only found this issue by randomly clicking around in the content_harvest mapped pages. Since 27143 is so large, with 1,020 mapped tasks, the task is prone to stop running several times along the way. I've been selecting "Clear task" > choosing the "Only Failed" option, then re-running the tasks to run only failed tasks.

```
*** Reading remote log from Cloudwatch log_group: airflow-pad-airflow-mwaa-Task log_stream: dag_id=harvest_collection/run_id=manual__2024-03-05T21_18_43+00_00/task_id=content_harvesting.content_harvest/map_index=749/attempt=2.log.
[2024-03-06, 02:44:27 UTC] {{taskinstance.py:1093}} INFO - Dependencies not met for <TaskInstance: harvest_collection.content_harvesting.content_harvest manual__2024-03-05T21:18:43+00:00 map_index=749 [success]>, dependency 'Task Instance State' FAILED: Task is in the 'success' state.
[2024-03-06, 02:44:27 UTC] {{local_task_job_runner.py:154}} INFO - Task is not able to be run
[2024-03-06, 02:54:42 UTC] {{taskinstance.py:1093}} INFO - Dependencies not met for <TaskInstance: harvest_collection.content_harvesting.content_harvest manual__2024-03-05T21:18:43+00:00 map_index=749 [success]>, dependency 'Task Instance State' FAILED: Task is in the 'success' state.
[2024-03-06, 02:54:42 UTC] {{local_task_job_runner.py:154}} INFO - Task is not able to be run
```

**Comment by barbarahui** [comment]
@christinklez It looks like the task that failed was `content_harvest` task 749. The `content_harvest` row in the grid view is displaying a red square as it should. Maybe I'm misunderstanding the problem. Let me know if you want to hop on zoom to clarify!

**Comment by barbarahui** [comment]
@christinklez Oh nvm, I see what you mean! I was looking the wrong grid!

**Comment by barbarahui** [comment]
@christinklez I think there _may_ be some confusion around the way Airflow handles running tasks. It will sometimes retry mapped tasks that have already been run (due to the parent task running for a very long time) and then report that it couldn't run them because they've already been run and have succeeded. For example, here's an excerpt from the message you linked to:

`dependency 'Task Instance State' FAILED: Task is in the 'success' state.`

It's very confusing, but I think Airflow is simply saying that that it won't run the task because it's already succeeded. And indeed the first attempt has succeeded.

Does that make sense and match your experience of things?

Also, when you say that "the task is prone to stop running several times along the way", what do mean you mean exactly? What do you see in the logs?

**Comment by christinklez** [comment]
Thanks @barbarahui! It's a bit tricky for me to figure out when a "failed" task has previously "succeeded," but I will watch out for that!

By "the task is prone to stop running several times along the way," -- this tends to happen with larger collections that have many task pages. Usually during the `map_page` or `content_harvest` task.

Sometime while it's running a batch of `map_page` or `content_harvest` task pages, it will stop running and turn red. It will have failed on a particular page (but it gets tricky to find the particular page). The entire job will then stop. Once stopped, I can retry the "failed tasks" only, and it will retry only the red task pages.

**Here's an example:**
- Run ID: manual__2024-03-27T00:02:50+00:00
- During content_harvest mapped page 8
   - [Link to airflow grid](https://7a8067cb-3b99-477e-a883-7e311175a9b4.c3.us-west-2.airflow.amazonaws.com/dags/harvest_collection/grid?num_runs=365&dag_run_id=manual__2024-03-27T00%3A02%3A50%2B00%3A00&task_id=content_harvesting.content_harvest&tab=mapped_tasks&map_index=8)
   - [Permalink to log](https://7a8067cb-3b99-477e-a883-7e311175a9b4.c3.us-west-2.airflow.amazonaws.com/log?dag_id=harvest_collection&task_id=content_harvesting.content_harvest&execution_date=2024-03-27T00%3A02%3A50%2B00%3A00&map_index=8)

In this example, you can see there were two attempts. The first attempt error'd out and failed (along with other pages that failed). Later, when the entire content_harvest job failed, I clicked on the content_harvest job (and not a specific task page) and clicked "Clear" > "Only Failed" to rerun the failed pages. The second attempt was successful.

**Comment by barbarahui** [comment]
@christinklez To clarify re figuring out when a "failed" task has previously "succeeded": what I mean is that the task has never actually failed! That's why the task is marked with a green square. The message you're seeing from Airflow means that the conditions for rerunning the task were not met, so it didn't rerun the task. So it's correct that the task is marked with a green square as being successful. I don't think you need to watch out for these. You can rely on the red squares.

**Comment by barbarahui** [comment]
@christinklez Re long-running tasks stopping, I think Airflow is running into a timeout limit and killing the workers at that time. I'll investigate what we can do to increase that limit so that you don't run into this.

**Comment by christinklez** [comment]
Likely fixed via #856
